### PR TITLE
Make service announcements a little nicer

### PR DESCRIPTION
**Pull request by JoshAshby**
Makes the icons and colors on service announcements into enums, adds a preview to the show page and make the dropdowns for the two all fancy and neat. Also turns the message into markdown and I'm going to make a switch so that they can be for logged in only users as well.

transientbug admin viewing test 2018-02-12 09-17-12

transientbug admin editing test 2018-02-12 09-22-43

transientbug admin editing test 2018-02-12 09-23-03

transientbug admin editing test 2018-02-12 09-23-27

transientbug admin editing test 2018-02-14 15-45-01

### CODE REVIEW

File: app/models/service_announcement.rb [hunk 1]
```diff
@@ -1,6 +1,32 @@
...
+    olive: 4,
+    green: 5,
+    teal: 6,
+    blue: 7,
+    violet: 8,
+    purple: 9,
+    pink: 10,
+    brown: 11,
+    grey: 12,
+    black: 13
```
Comment (robacarp) [inline app/models/service_announcement.rb]: dat enum

File: db/migrate/20180212151406_change_service_announcement_icon_to_enum.rb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+class ChangeServiceAnnouncementIconToEnum < ActiveRecord::Migration[5.2]
+  def change
+    rename_column :service_announcements, :icon, :icon_text
+    add_column :service_announcements, :icon, :integer, default: 0
```
Comment (robacarp) [inline db/migrate/20180212151406_change_service_announcement_icon_to_enum.rb]: >integer

thats not an enum

File: app/models/service_announcement.rb [hunk 3]
```diff
@@ -1,6 +1,32 @@
 class ServiceAnnouncement < ApplicationRecord
-  validates :color, inclusion: {
-    in: %w( plain red orange yellow olive green teal blue violet purple pink brown grey black )
+  # https://semantic-ui.com/globals/site.html#colors
+  enum color: {
+    plain: "plain",
```
Comment (JoshAshby) [inline app/models/service_announcement.rb]: @robacarp fixed >:D

File: db/migrate/20180212151406_change_service_announcement_icon_to_enum.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+class ChangeServiceAnnouncementIconToEnum < ActiveRecord::Migration[5.2]
+  def up
+    ActiveRecord::Base.connection.execute <<~SQL
+      CREATE TYPE icon AS ENUM ('announcement', 'help', 'info', 'warning', 'talk', 'settings', 'alarm', 'lab');
```
Comment (JoshAshby) [inline db/migrate/20180212151406_change_service_announcement_icon_to_enum.rb]: Now it is @robacarp

File: db/schema.rb [hunk 5]
```diff
@@ -192,17 +192,8 @@
...
-    t.text "color"
-    t.datetime "start_at"
-    t.datetime "end_at"
-    t.datetime "created_at", null: false
-    t.datetime "updated_at", null: false
-    t.text "icon"
-    t.boolean "active", default: true
-  end
+# Could not dump table "service_announcements" because of following StandardError
+#   Unknown type 'icon' for column 'icon'
```
Comment (JoshAshby) [inline db/schema.rb]: @robacarp WELP

Comment (robacarp) [reply db/schema.rb]: Looks like you’re going to have to extend activerecord to manage that

Comment (JoshAshby) [reply db/schema.rb]: don't say that, I will

Comment (JoshAshby) [reply db/schema.rb]: Oh good someone already did: https://gist.github.com/clarkdave/5936375

brb, converting to structure.sql I guess :/

**Pull request merged**